Graphics/GraphicsEngineD3D11/src/ShaderResourceCacheD3D11.cpp

```diff
@@ -217,6 +207,21 @@ void ShaderResourceCacheD3D11::TransitionResources(DeviceContextD3D11Impl& Ctx,
         }
     }
 }
+    
+template<ShaderResourceCacheD3D11::StateTransitionMode Mode>
+void ShaderResourceCacheD3D11::TransitionResourceStates(DeviceContextD3D11Impl& Ctx)
+{
+    VERIFY_EXPR(IsInitialized());
+    
+    TransitionResources<Mode>(Ctx, static_cast<ID3D11Buffer*>(nullptr));
+    TransitionResources<Mode>(Ctx, static_cast<ID3D11ShaderResourceView*>(nullptr));
+    TransitionResources<Mode>(Ctx, static_cast<ID3D11SamplerState*>(nullptr));
+    TransitionResources<Mode>(Ctx, static_cast<ID3D11UnorderedAccessView*>(nullptr));
+}
+
+// Instantiate templates
+template void ShaderResourceCacheD3D11::TransitionResourceStates<ShaderResourceCacheD3D11::StateTransitionMode::Transition>(DeviceContextD3D11Impl& Ctx);
+template void ShaderResourceCacheD3D11::TransitionResourceStates<ShaderResourceCacheD3D11::StateTransitionMode::Verify>(DeviceContextD3D11Impl& Ctx);

```

Comment:
Please don't move the function in the file - it only creates extra delta

Reply:
It doesn't compile if put into the .hpp due to the definition only being in .cpp.
No idea why it works in MSVC.

Reply:
I mean, you moved the `ShaderResourceCacheD3D11::TransitionResourceStates` function as is in the `ShaderResourceCacheD3D11.cpp` file, which created extra changes.

Reply:
Oh, let me move it back to where it was.

Reply:
Moved back.